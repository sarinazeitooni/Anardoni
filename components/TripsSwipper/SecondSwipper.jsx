import React, {useEffect, useState} from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import style from './scss/trips-swipper.module.scss';
import messages from "./messages/messages";
import {useSelector, useDispatch} from "react-redux";
import SwiperSkeleton from "../skeleton/SwiperSkeleton";
import SmallCardContainer from "../smallCard/SmallCardContainer";
import swiperAction from "../../redux/action/Swiperaction";

function SecondSwipper({title}) {
    const [swipperData, setSwipperData] = useState(null);
    const dispatch = useDispatch();
    const swipperDataForTimeOut = useSelector(state => state.swipper);
    useEffect(() => {
        dispatch(swiperAction());
        setTimeout(() => {
            setSwipperData(swipperDataForTimeOut);

        }, 1000);
    }, [swipperDataForTimeOut]);
    return (
        <div className={style['second-swipper-container']}>
            <div className={style['swipper-title']}>
                <div className={style['right-title']}>{title}</div>
                <div className={style['left-title']}><span>{messages.more}</span></div>
            </div>
            <Swiper spaceBetween={0} slidesPerView={7}>
                {!swipperData && (
                    <SwiperSkeleton slides={10}/>
                )}
                {swipperData && (swipperData.map((item) => {
                    return (
                        <SwiperSlide key={item.id}>
                            <SmallCardContainer title={item.title} detail={item.detail} cardImage={item.url}/>
                        </SwiperSlide>
                    )
                }))
                }
            </Swiper>
        </div>)
}
export default SecondSwipper;