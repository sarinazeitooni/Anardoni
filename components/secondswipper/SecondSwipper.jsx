import React , {useState} from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import SmallCardContainer from "../smallCard/SmallCardContainer";
import {BrowserView, MobileView} from "react-device-detect";
import { useSelector , useDispatch } from "react-redux";
import MySkeleton from "../skeleton/Skeleton";
function SecondSwipper(){
    const [swipperData , setSwipperData] = useState(null);
    const dispatch = useDispatch();
    dispatch({
        type: "swipper"

    });
    const swipperDataForTimeOut = useSelector(state => state.swipper);
    setTimeout(()=>{
        setSwipperData(swipperDataForTimeOut);
    },3000)
    return (
        <div className={'second-swipper-container'}>
            <div className={"swipper-title"}>
                <div className={'right-title'}>سفرهای ضروری</div>
                <div className={'left-title'}>بیشتر</div>
            </div>
            <BrowserView>
                <Swiper spaceBetween={0} slidesPerView={8}>
                    {!swipperData && (
                        <MySkeleton slides={8}/>
                    )}
                    {swipperData && (swipperData.map((item)=>{
                        return(
                            <SwiperSlide>
                                <SmallCardContainer title={item.title} detail={item.detail} cardImage={item.url} />
                            </SwiperSlide>
                        )
                    }))
                    }
                </Swiper>
            </BrowserView>
            <MobileView>
                <Swiper spaceBetween={0} slidesPerView={2}>
                    {!swipperData && (
                        <MySkeleton slides={2}/>
                    )}
                    {swipperData && (swipperData.map((item)=>{
                        return(
                            <SwiperSlide>
                                <SmallCardContainer title={item.title} detail={item.detail}
                                                    cardImage={item.url} />
                            </SwiperSlide>
                        )
                    }))
                    }
                </Swiper>
            </MobileView>
        </div>)
};
export default SecondSwipper;