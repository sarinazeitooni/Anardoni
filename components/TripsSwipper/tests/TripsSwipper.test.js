import React from 'react';
import { shallow } from 'enzyme';
import SecondSwipper from "../TripsSwipper";
import store from "../../../redux/store/store";
import {Provider} from 'react-redux';
describe(" Trip swiper component", () => {
    it("should have a swiper", () => {
        const swiper = shallow(<Provider store={store}> <SecondSwipper/></Provider>);
        expect(swiper).toMatchSnapshot();
    });
    it("it should render a preview" ,()=>{
        const swiper = shallow(<Provider store={store}> <SecondSwipper/></Provider>);
        expect(swiper.find('div.more-container')).toBeTruthy();
    });
});