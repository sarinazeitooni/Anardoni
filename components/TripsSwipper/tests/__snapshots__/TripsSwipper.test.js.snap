// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Trip swiper component should have a swiper 1`] = `ShallowWrapper {}`;
